packages/cloud-functions/src/circuitBreaker/circuitBreaker.ts

```diff
@@ -0,0 +1,89 @@
+import * as functions from 'firebase-functions'
+import { KeyManagementServiceClient, protos } from '@google-cloud/kms'
+import express from 'express'
+import { CIRCUIT_BREAKER_KEY_PATH } from '../config'
+
+type CryptoKeyVersionState = keyof typeof protos.google.cloud.kms.v1.CryptoKeyVersion.CryptoKeyVersionState
+
+const client = new KeyManagementServiceClient()
+
+const app = express()
+app.use(express.json())
+
+async function getKeyState(): Promise<CryptoKeyVersionState> {
+  const getCryptoKeyVersionResult = await client.getCryptoKeyVersion({
+    name: CIRCUIT_BREAKER_KEY_PATH,
+  })
+  return getCryptoKeyVersionResult[0].state as CryptoKeyVersionState
+}
+
+app.get('/health', async (req, res) => {
+  return res.status(200).json({
+    status: 'ok',
+  })
+})
+
+app.get('/status', async (req, res) => {
+  const keyState = await getKeyState()
+  switch (keyState) {
+    case 'ENABLED':
+      return res.status(200).json({
+        status: 'enabled',
+      })
+    case 'DISABLED':
+      return res.status(200).json({
+        status: 'temporarily disabled',
+      })
+    case 'DESTROYED':
+      return res.status(200).json({
+        status: 'permanently disabled',
+      })
+    default:
+      console.error(`Unexpected key state: ${keyState}`)
+      return res.status(200).json({
```

Comment:
200 -> 400?

Reply:
not really a 400 though (imo); the request itself is fine, we're just getting an unexpected state for the key internally (i.e., a dev probably manually changed it to something other than `enabled`, `disabled`, or `destroyed`.)

Reply:
ah, makes sense.